README.md

```diff
@@ -17,34 +18,44 @@ Your charm directory should have the following overall structure:
 |   +-- charm.py
 +-- hooks/
     +-- install -> ../src/charm.py
+    +-- start -> ../src/charm.py  # for k8s charms per below
 ```
 
-The `mod/` directory will contain the operator framework dependency as a git
+The `mod/` directory should contain the operator framework dependency as a git
 submodule:
 
 ```
 git submodule add https://github.com/canonical/operator mod/operator
 ```
 
-Other dependencies included as git submodules can be added there as well.
+You can sync subsequent changes from the framework and other submodule
+dependencies by running:
+
+```
+git submodule update
+```
 
-The `lib/` directory will then contain symlinks to subdirectories of your
-submodule dependencies to enable them to be imported into the charm:
+Then symlink from the git submodule for the operator framework into the `lib/`
+directory of your charm so it can be imported at run time:
 
 ```
 ln -s ../mod/operator/ops lib/ops
 ```
 
+Other dependencies included as git submodules can be added in the `mod/`
+directory and symlinked into `lib/` as well.
+
 Your `src/charm.py` is the entry point for your charm logic. It should be set
-to executable and use Python 3.6 or greater. At a minimum, it needs to define a
-subclass of `CharmBase` and pass that into the framework's `main` function:
+to executable and use Python 3.6 or greater (as such, operator charms can only
+support Ubuntu 18.04 or later). At a minimum, it needs to define a subclass of
+`CharmBase` and pass that into the framework's `main` function:
 
 ```python
 import sys
-sys.path.append('lib')
 
-from ops.charm import CharmBase
-from ops.main import main
+from lib.ops.charm import CharmBase
+from lib.ops.main import main
```

Comment:
I don't think this actually works. The CWD when executing a charm is set to the charm root directory, however Python doesn't put "." into sys.path by default. It does put the directory of the executable.
We could recommend that people put the symlink for 'ops' into src/ next to charm.py if we wanted to avoid sys.path manipulation.
sys.path.append('lib') is also sensitive to anyone doing an os.chdir(), as then the relative path of 'lib' actually points to a different directory.
I know there is a desire to clearly identify what is "my charm code" vs "what are my dependencies". Which is what initiated the desire for a 'lib' directory in the first place. (Also because you can't submodule the ops directory directly into charm/* because then the path is operator/ops.)